hbase-client/src/main/java/org/apache/hadoop/hbase/client/RpcRetryingCallerImpl.java

```diff
@@ -95,16 +104,26 @@ public T callWithRetries(RetryingCallable<T> callable, int callTimeout)
     List<RetriesExhaustedException.ThrowableWithExtraContext> exceptions = new ArrayList<>();
     tracker.start();
     context.clear();
+
+    OperationInterceptor operationInterceptor =
+      operationInterceptorFactory.createInterceptor();
+
     for (int tries = 0;; tries++) {
       long expectedSleep;
+
       try {
         // bad cache entries are cleared in the call to RetryingCallable#throwable() in catch block
         callable.prepare(tries != 0);
         interceptor.intercept(context.prepare(callable, tries));
```

Comment:
Does this interceptor not provide what you need? Looks more limited but does provide pre attempt and post failure hooks. I see that it's private though and doesn't seem very friendly